### DISCUSSION TITLE
Viewscreen

### DISCUSSION THREAD

**Discussion by Xansta** [Thorium-Sim/thorium-nova#41] [General]
From #21 

### Thorium Nova Core

**Viewscreen**

- Question: Should the main view always be the star map, or should it be able to show other things, like videos? How much like Thorium Classic?
- Question: What kinds of overlays will be included? (HUD-style)

We should probably define a viewscreen. From Star Trek and other shows, I believe it is the big screen at the front of the bridge. I've heard it called screen, forward screen, front screen and main screen. 

Artemis and Empty Epsilon allow for a 3D exterior view (front, back, left, right) with or without an exterior shot of the ship. There's also an option for a longer range view of the star field. Both Artemis and Empty Epsilon present this as an overhead view essentially in two dimensions. Artemis shows the entire 5x5 area. EE shows a circular representation of the reach of the long range sensors. Both views have representations for stellar phenomena like nebulae, black holes, planets, etc. Both views have icons in motion for ships. Both have representations for stations where the ship can dock. Artemis and EE also have a tactical view that can be placed on the forward screen. This is a shorter range view designed as an aid to combat. Artemis offers a data view that can be placed on the main screen which contains text readouts for weapons stocks, energy, etc. along with a rotating view of damaged systems similar to what's available to the Engineer.

Empty Epsilon has some auxiliary screens available. One is s ship window which provides an exterior 3D view at the angle specified. This is useful if you have spare monitors for the sides of your physical bridge set: set the one on the right to 90 degrees and the one on the left to 270 degrees and you can always "look out the window" to see what is happening on either side. There is also a cinematic view where an algorithm chooses the camera angle to view various ships in motion or in combat. This is useful to put outside the physical bridge set to keep observers somewhat entertained while they wait their turn. 

Artemis offers the ability to inject videos on the main screen. I have not used this feature, but I think it could be useful for Nova.

So, to propose an answer to the question regarding the content of the viewscreen (assuming it is also known as the main screen, forward screen, front screen, etc.): I think the captain should be able to select the content. Available content should be in priority order:
1) Exterior 3D view
2) Long range map
3) Short range map
4) Short range map with dynamic elements in motion
5) Long range map with dynamic elements in motion 
6) Ship map summary highlighting damaged systems (maybe deck by deck if necessary)
7) Ad hoc content like communication videos, static picture of alien or faction representative while audio plays (or is voiced by the GM), faction relationship diagram, planetary details retrieved from science database, etc.

If we're modeling everything internally, the chances are good we can make a visual model corresponding to the internal model suitable for the main screen.

Considering the items listed, perhaps the captain and the flight director share control of the main screen. For example, maybe the captain can readily switch between the exterior view and the long range map. The flight director may know when a transmission is coming in from another ship that has a corresponding visual representation or not - maybe it's just audio or maybe the communication is text only. Perhaps there could be an option to put up the Relay or Communication officer screen when messages are coming in from external sources.

**Comment by alexanderson1993** [comment]
I agree with most of these. A few specific points:

> If we're modeling everything internally, the chances are good we can make a visual model corresponding to the internal model suitable for the main screen.

I was able to accomplish this with Thorium Nova Prime. All of the viewscreens you list should be technically doable.

>  perhaps the captain and the flight director share control of the main screen

Agreed, along with automatically changing the viewscreen as part of a mission timeline.

Attached are two images from Star Trek: Beyond showing the Enterprise viewscreen.

![WA_OO_ViewScreen_4_001e](https://user-images.githubusercontent.com/6558157/137174992-2f21d0c9-dba4-4d9e-8ec0-c6cc9ce48384.jpg)

![WA_OO_ViewScreen_3_002a](https://user-images.githubusercontent.com/6558157/137175007-2a9e1254-ba05-4bda-a2b7-95d8cdbeb6c8.jpg)

Notice in the second image the items along the side continuously showing data about the ship and the surrounding area. This is what I meant by "What kinds of overlays will be included?"

In the first image, it shows things _on top_ of the outside view. I think these are best suited for specific alerts, like "Collision Alert" or "Intruder Alert". I also think it should be possible for other people to place information from their stations on the viewscreen based on context. For example, communications putting up the current short-range communication call, sensors putting up the scan results of a ship or planet, etc. And, of course, the Flight Director or Mission Timeline can override what's shown on the viewscreen at any time, and the Captain can clear it entirely if the outside view is obscured.

As an aside, there's really not much to see on the viewscreen at real-world sizes and distances, since most everything is so dang far away and so dang small. Most likely, instead of seeing actual ships, the viewscreen will show icons with numbers indicating the distance unless the ship is close enough that you can actually make out the 3D model.

My goal with the viewscreen is to make it as automated as possible. It automatically responds to the simulation to show what is most important, it automatically displays alerts and whatnot. Thorium Classic takes a lot of effort for the Flight Director to set up the viewscreen, which is why I want Nova to automate it so much.

I think there should be a separate discussion sometime in the future about how the short and long range maps work, specifically with what they show, how the crew controls the map, whether the crew is able to place things on the map or draw on it, etc.

**Reply by alexanderson1993** [reply]
Let me clarify - I show those images as a broad illustration. To be frank, that's crap user interface design, meant to be showy for a movie, but wholly unhelpful to an actual crew. Our viewscreen should avoid superfluous frills and focus on real, helpful information.

**Reply by astrolamb-gaming** [reply]
Some basic overlays could be heading, target indicators relative to current bearing, general status indicators for various systems (i.e. fuel/energy reserves), basic damage indicators, and weapons/shield statuses.

**Reply by Xansta** [reply]
"Magnify"
"Aye, magnification factor 10"

**Reply by alexanderson1993** [reply]
Yes to all of this!

**Reply by Xansta** [reply]
I can see sets of HUDs. For combat (red alert?), you'd want shield status, weapons status, targeting reticles, hull status. For peaceful portions of the mission, perhaps Energy status, Crew readiness state, Watch status (morning, forenoon, afternoon, first dog, second dog, first, middle, etc.), maintenance tasks status, etc.

Of course, you can have a custom set of HUDs for each scenario/mission/timeline. I recommend we define a default set for each alert level. 

Speaking of alert levels, The Artemis and Empty Epsilon alerts leave something to be desired. Specifically, they distract from a console operators' duties. If we have something like normal, yellow and red alerts, I suggest that any major graphic change be temporary in nature followed by a small indicator. Also, in Artemis and Empty Epsilon, the alert level has no effect on game play. I believe that Starship Horizons has implemented a more practical application, something along the lines of you can't load torpedo tubes or raise shields until you are at yellow alert and you can't fire weapons unless you're at red alert.

**Reply by alexanderson1993** [reply]
> Speaking of alert levels

I propose 3 alert levels. Thorium Classic has 5, and I grew to hate it. For one, Alert Level 4 being green is just gross gross gross. And its overly complicated. 3 alert levels is more than enough for our purposes.

> Starship Horizons has implemented a more practical application,

I appreciate the sentiment, but that actually annoys me about Starship Horizons. It made it a pain to figure out how to launch when I first started, and it's not clear what each alert level allows and disallows.

I think there are ways we could make alert conditions affect gameplay without being too restrictive, or at least by explaining why certain actions can't be taken while in a certain alert condition via a tooltip or notification.

> I suggest that any major graphic change be temporary in nature followed by a small indicator

This will be defined on a theme-by-theme basis, and since themes can be user-generated, there's not much control we have over those. I think we can be thoughtful about how we design themes. I agree - less is more when it comes to this kind of thing.

Just because I can, here are screenshots of most of the current Thorium layouts/themes in red alert, including the gratuitous ripoff of Empty Epsilon, courtesy BlueShadow. ;)
 
![localhost_4442_client](https://user-images.githubusercontent.com/6558157/137209191-a2225946-8d2d-469f-9903-69c7f89dbe79.jpg)
![localhost_4442_client (2)](https://user-images.githubusercontent.com/6558157/137209194-ce93b998-7555-4ef0-82ab-eb16bc32e4ed.jpg)
![localhost_4442_client (3)](https://user-images.githubusercontent.com/6558157/137209198-52cf2d3b-ddba-441f-8996-adb887d12fb6.jpg)
![localhost_4442_client (4)](https://user-images.githubusercontent.com/6558157/137209201-d97c19e7-b8a6-4b79-a2e1-746e10105c90.jpg)
![localhost_4442_client (5)](https://user-images.githubusercontent.com/6558157/137209204-2d649c5d-ac4a-43f8-a848-748b01793c43.jpg)
![localhost_4442_client (6)](https://user-images.githubusercontent.com/6558157/137209207-f87b73aa-5357-452a-b0c9-f842f3c7f7a1.jpg)
![localhost_4442_client (7)](https://user-images.githubusercontent.com/6558157/137209209-67d7d7a3-a3c6-4526-bcad-8fbe9788d73a.jpg)
![localhost_4442_client (8)](https://user-images.githubusercontent.com/6558157/137209212-dd57ea5e-f895-44c9-9651-8ee304de7762.jpg)
![localhost_4442_client (1)](https://user-images.githubusercontent.com/6558157/137209216-b0c5e5d2-5f3c-42f1-9592-5e779e0d3936.jpg)

**Reply by aBlueShadow** [reply]
> For one, Alert Level 4 being green is just gross gross gross. 

Which is why I swapped the colors of 4 and 5 for the Epsilon theme :-)
So 4 is blue alert*, which is kinda fitting, as blue alert in star trek is some exceptional condition that need more attention than usual, often without being necessarily harmful, like landing maneuvers or entering a space dock.

_*At least on the status screen, seems that I missed that alert card_
